File: WorkManagerDemo/app/src/main/java/com/androidtutz/anushka/workmanagerdemo/MainActivity.java
package com.androidtutz.anushka.workmanagerdemo;

import android.databinding.DataBindingUtil;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.Toolbar;
import android.view.Menu;
import android.view.MenuItem;
import android.widget.Toast;

import androidx.work.Constraints;
import androidx.work.Data;
import androidx.work.OneTimeWorkRequest;
import androidx.work.WorkManager;

import com.androidtutz.anushka.workmanagerdemo.databinding.ActivityMainBinding;

public class MainActivity extends AppCompatActivity {

    public final static String KEY_COUNT_VALUE = "key_count";
    public final static String KEY_RETURN_VALUE = "key_return";

    private ActivityMainBinding binding;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = DataBindingUtil.setContentView(this, R.layout.activity_main);
        setSupportActionBar(binding.toolbar);

        Data data = new Data.Builder()
                .putInt(KEY_COUNT_VALUE, 30000)
                .build();

        Constraints constraints = new Constraints.Builder()
                .setRequiresCharging(true)
                .build();

        final OneTimeWorkRequest request = new OneTimeWorkRequest.Builder(DemoWorker.class)
                .setInputData(data)
                .setConstraints(constraints)
                .build();

        binding.fab.setOnClickListener(view -> WorkManager
                .getInstance().enqueue(request));

        WorkManager.getInstance()
                .getWorkInfoByIdLiveData(request.getId())
                .observe(this, workInfo -> {
                    if(workInfo != null) {
                        binding.content.tvStatus.setText(workInfo.getState().name());

                        if(workInfo.getState().isFinished()) {
                            Data data1 = workInfo.getOutputData();

                            Toast.makeText(this, data1.getString(KEY_RETURN_VALUE),
                                    Toast.LENGTH_LONG).show();
                        }
                    }
                });
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        getMenuInflater().inflate(R.menu.menu_main, menu);
        return true;
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        int id = item.getItemId();

        if (id == R.id.action_settings) {
            return true;
        }

        return super.onOptionsItemSelected(item);
    }
}

File: ActivityStatesDemo/app/src/main/java/com/example/activitystatesdemo/view/MainActivity.kt
package com.example.activitystatesdemo.view

import android.content.Intent
import android.os.Bundle
import android.view.Menu
import android.view.MenuItem
import androidx.appcompat.app.AppCompatActivity
import androidx.databinding.DataBindingUtil
import com.example.activitystatesdemo.R
import com.example.activitystatesdemo.component.DemoAppComponent
import com.example.activitystatesdemo.databinding.ActivityMainBinding
import com.example.activitystatesdemo.utils.log

class MainActivity : AppCompatActivity() {

    private lateinit var binding: ActivityMainBinding
//    private val activityName = this::class.java.simpleName

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = DataBindingUtil.setContentView(this,R.layout.activity_main)
        setSupportActionBar(binding.toolbar)

        binding.fab.setOnClickListener {
            startActivity(Intent(this, SecondActivity::class.java))
        }

        lifecycle.addObserver(DemoAppComponent(this::class.java.simpleName))
    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        menuInflater.inflate(R.menu.menu_main, menu)
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        return when (item.itemId) {
            R.id.action_settings -> true
            else -> super.onOptionsItemSelected(item)
        }
    }
}
